x-pack/qa/repository-old-versions-7x/build.gradle

```diff
@@ -0,0 +1,9 @@
+import org.elasticsearch.gradle.internal.test.RestIntegTestTask
+import org.elasticsearch.gradle.Version
+
+apply plugin: 'elasticsearch.internal-java-rest-test'
+
+tasks.named("javaRestTest").configure {
+  usesDefaultDistribution()
+  usesBwcDistribution(Version.fromString("7.17.25"))
```

Comment:
@breskeby that looks simple enough, but going forward I would need several "old" clusters (i.e. 7.0.0, maybe even some other minor 7x versions where the Lucene codecs changed) for the snapshot data setup and then they would need to run against a clean version of the "current" cluster version for the tests.

I'm wondering if I should use something like https://github.com/elastic/elasticsearch/blob/1d4c8d85f6641f8f4efa776106392b0eb6980406/qa/rolling-upgrade/build.gradle#L20 where we seem to use the `bwcVersion` probably coming from the `elasticsearch.bwc-test` plugin or if we can supply a "fixed" list of versions we want to run as "old" versions here. Would I need to register individual tasks for each "old" version?

Also, does this definition here already "start" the clusters or is that happening later with  `ElasticsearchCluster.local()[...].build()` in the test?

Reply:
I'd go ahead with a fixed list as we did for other "old elasticsearch" tests. 
I think the versions you wanna use are listed in that task. That probably needs some tweaking of the RestTestBasePlugin to be supported. 

This definition basically only downloads or builds the distribution you wanna use in your tests. they are only started in `ElasticsearchCluster.local()[...].build()`. Given that. I think we should only have one test task here and do the parameterisation of the versions we test in the test code. e.g. having a parameterised junit test per version under test. the lifecycle of the "current" cluster for that would also live in the test code then. 

One problem we might see (haven't completely checked) is that those older cluster only support a limited range of java versions? we probably need to ensure those clusters are wired to the correct java version when starting up.

Reply:
> Given that. I think we should only have one test task here and do the parameterisation of the versions we test in the test code. e.g. having a parameterised junit test per version under test.

I'm not sure I understand. Do you propose to start all "old version" clusters consecutively from only one test suite? That would make it hard to e.g. from gradle only run the tests for e.g 7.0.0 vs. CURRENT, or am I missunderstanding something? Or do you mean we pass in the "old version" we are running for the data setup to the test e.g. via a system property and loop over it somewhere in the build.gradle definition? I would like to avoid having to run all old version tests at once every time, at the same time having the flexibility to use more than one test suite with basically the same fixed setup (ideally the old data setup is just done once).